src/Sylius/Bundle/ApiBundle/spec/Serializer/TranslatableDenormalizerSpec.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Serializer;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Resource\Model\TranslatableInterface;
+use Sylius\Component\Resource\Translation\Provider\TranslationLocaleProviderInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class TranslatableDenormalizerSpec extends ObjectBehavior
+{
+    function let(
+        DenormalizerInterface $denormalizer,
+        TranslationLocaleProviderInterface $localeProvider,
+    ): void {
+        $this->beConstructedWith($localeProvider);
+
+        $this->setDenormalizer($denormalizer);
+    }
+
+    function it_only_supports_translatable_resource_with_no_translations(): void
```

Comment:
```suggestion
    function it_only_supports_translatable_resource(): void
```
as there is no check in `supportsDenormalization`, if there are any translations in the given data 🤔